helm/resource_release.go

```diff
@@ -429,6 +434,15 @@ func resourceRelease() *schema.Resource {
 					},
 				},
 			},
+			"upgrade": {
+				Type:        schema.TypeMap,
```

Comment:
The pattern we usually follow is to have this kind of schema construct be a `TypeList` with a single element `Elem: &schema.Resource{...` which contains needed the sub-attributes. There are historical reasons for this approach rooted in limitations of the provider SDK.

In short, can you please refactor this block along the lines of "postrender" or "metadata" above?

Reply:
Thank you for the feedback, sorry it took me a bit to see this-- been a mad few days.  I'll get this sorted out.